client/src/elements/item-preview/preview-container.js

```diff
@@ -24,15 +20,45 @@ export class PreviewContainer {
     this.qConfig = qConfig;
     this.id = `preview-container-${Math.floor(Math.random() * 10 ** 9)}`;
     this.element.setAttribute("id", this.id);
+    this.currentScaleFactor = 1;
   }
 
   async attached() {
     this.QServerBaseUrl = await qEnv.QServerBaseUrl;
+    this.initScaleToFit();
     this.showPreview(this.renderingInfo);
   }
 
+  async detached() {
+    this.unobservePreviewResize();
+  }
+
+  widthChanged(width) {
+    if (!width) {
+      return;
+    }
+    if (Number.isNaN(parseFloat(width))) {
```

Comment:
What was the intention of that method? `Number.isNaN(parseFloat('100%'))` never lands in here as `parseFloat('100%')` results in 100. You probably wanted to use `Math.floor()`.

I'd use 
```js
number = parseFloat(width)
if (width === 'auto') this.cssWidth = '100%' 
else if (width.endsWith('%')) this.cssWidth = `${number}%`
else if (width.endsWith('px')) this.cssWidth = `${number}px`
else throw new Error(`Unsupported width value '${width}'`)
```

Reply:
Ah yes, my code is very confusing. I probably had no clear intention as far as it looks.
I was really not sure if I wanted to implement the support for units defined in the option. Before only a Number that was then used with the px unit was supported and I didn't really intent to support more than that besides `auto`.

I am still not sure about that so I would not open the possibility just now, also because there is still the problem of the misleading naming: `value` instead of `width` in the config.

Changing the config format for the preview in the future could be a good reason to the really think about supporting units in there.

Do you agree if I'd go with the `Number(width)` check for now?

Reply:
Yes, supporting number is totally fine. But I'd put the `Unsupported width value` error into  another else clause.

Reply:
c9dd128a6e94ac6df06f3b93cb50917b5c480229